File: JANUARY/24-01-2026/1877.php
class Solution {
    function minPairSum($nums) {
        sort($nums);
        $i = 0;
        $j = count($nums) - 1;
        $ans = 0;
        while ($i < $j) {
            $ans = max($ans, $nums[$i] + $nums[$j]);
            $i++;
            $j--;
        }
        return $ans;
    }
}

File: JANUARY/24-01-2026/1877.rs
impl Solution {
    pub fn min_pair_sum(mut nums: Vec<i32>) -> i32 {
        nums.sort();
        let mut i = 0usize;
        let mut j = nums.len() - 1;
        let mut ans = 0;
        while i < j {
            ans = ans.max(nums[i] + nums[j]);
            i += 1;
            j -= 1;
        }
        ans
    }
}